include/envoy/stats/stats.h

```diff
@@ -20,39 +21,90 @@ class Instance;
 namespace Stats {
 
 /**
+ * General representation of a tag.
+ */
+struct Tag {
+  std::string name_;
+  std::string value_;
+};
+
+class TagExtractor {
+public:
+  virtual ~TagExtractor() {}
+
+  /**
+   * Identifier for this tag.
+   */
+  virtual std::string name() const PURE;
+
+  /**
+   * Updates the tag extracted name and the set of tags by extracting the tag represented by this
+   * TagExtractor. If the tag is not represented in the current tag_extracted_name, nothing will be
```

Comment:
The literal `tag_extracted_name` now no longer is a thing in this comment, so all references to it should be updated. Also, it's not really an "update" operation anymore.